avocado/plugins/run.py

```diff
@@ -89,6 +91,10 @@ def configure(self, parser):
         parser.add_argument('--ignore-missing-references', choices=('on', 'off'),
                             help="Force the job execution, even if some of "
                             "the test references are not resolved to tests.")
+        parser.add_argument("--dump-variants", default=None,
+                            help="Dump the Variants to a JSON file")
+        parser.add_argument("--load-variants", default=None,
+                            help="Load the Variants from a JSON file")
```

Comment:
This is tricky, because it has the same issue as `--replay-ignore mux`. One might expect the loaded variants would be combined with the newly defined ones (which is not happening). Replay warns the user about this and I'm wondering whether we should do the same here.

Reply:
Actually with replay it's even more confusing:

```
avocado run --replay d3aa78d --load-variants mux-0.json 
Using src job Mux data only, use `--replay-ignore variants` to override them.
```
but it uses the variants from `--load-variants`.

I see 3 options:

1. The current (with warning in -h, maybe even on execution, not sure how important this is)
2. The current but adding detection of variants. We failed doing that before, but we can re-try. In case variants are detected together with `--load-variants` it should warn user about ignored variants.
3. Implementing this as a varianter plugin, therefor it'd simply add the loaded variants to already specified ones (therefor `-m foo.yaml --load-variants bar.json` will yield all variants from `foo.yaml` as well as from `bar.json`.

I'd probably favor (3) as it should be fairly simple and will allow best of all worlds.